### PR TITLE
Update encryption-at-rest task page

### PR DESCRIPTION
**Pull request by sftim**
These changes are split out from PR #33285

- List available API encryption providers
  - The heading makes it clear you have a choice as a cluster operator about which to use
  - Provide better context about which to pick
- Promote creating the encryption configuration file to have its own page heading (and section)

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 1]
```diff
@@ -279,31 +282,44 @@ The following table describes each available provider:
...
+encrypts that data key using the managed encryption service. Kubernetes generates a unique data key for
+each resource. The API server stores an encrypted version of the data key in etcd alongside the ciphertext;
+when reading the resource, the API server calls the managed encryption service and provides both the
+ciphertext and the (encrypted) data key.
+Within the managed encryption service, the provider use a _key encryption key_ to decipher the data key,
+deciphers the data key, and finally recovers the plain text. Communication between the control plane
+and the KMS requires in-transit protection, such as TLS.
 
-## Encrypting your data
+Using envelope encryption creates dependence on the key encryption key, which not stored in Kubernetes.
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:308]:
```suggestion
Using envelope encryption creates dependence on the key encryption key, which is not stored in Kubernetes.
```

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 2]
```diff
@@ -279,31 +282,44 @@ The following table describes each available provider:
...
+In the KMS case, an attacker who intends to get unauthorised access to the plaintext data held at rest
+faces obstacles. In order to retrieve the plaintext values they would need to compromise etcd **and**
+the third-party KMS provider.
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:311]:
```suggestion
In the KMS case, an attacker who intends to get unauthorised access to the plaintext
values would need to compromise etcd **and** the third-party KMS provider.
```

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 3]
```diff
@@ -279,31 +282,44 @@ The following table describes each available provider:
 </tbody>
 </table>
 
-Each provider supports multiple keys - the keys are tried in order for decryption, and if the provider
-is the first provider, the first key is used for encryption.
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:283]: This content is removed?

Comment (sftim) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:283]: I'll move it to appear earlier in the page.

**Pull request merged**